File: packages/storage/ios/Plugin/Classes/Results/GetDownloadUrlResult.swift
import Foundation
import Capacitor

@objc public class GetDownloadUrlResult: NSObject, Result {
    private var url: URL

    init(url: URL) {
        self.url = url
    }

    public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["downloadUrl"] = url.absoluteString
        return result as AnyObject
    }
}

File: packages/storage/android/src/main/java/app/capgo/capacitor/firebase/storage/GetDownloadUrlResult.java
package app.capgo.capacitor.firebase.storage.classes.results;

import android.net.Uri;
import app.capgo.capacitor.firebase.storage.interfaces.Result;
import com.getcapacitor.JSObject;

public class GetDownloadUrlResult implements Result {

    private Uri uri;

    public GetDownloadUrlResult(Uri uri) {
        this.uri = uri;
    }

    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("downloadUrl", uri.toString());
        return result;
    }
}
